domain/machine/state/state.go

```diff
@@ -185,3 +185,34 @@ func (st *State) GetMachineLife(ctx context.Context, machineId string) (*life.Li
 	})
 	return &lifeResult, errors.Annotatef(err, "getting life status for machines %q", machineId)
 }
+
+// ListAllMachines retrieves the ids of all machines in the model.
+func (st *State) ListAllMachines(ctx context.Context) ([]string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	query := `SELECT &M.machine_id FROM machine`
```

Comment:
This type seems wrong and expensive on memory for no reason. Above we have introduced a type of instance_id. Here we should introduce machine id that way we get the exact memory we need.